profiler/src/ProfilerEngine/Datadog.Profiler.Native/FrameStore.cpp

```diff
@@ -932,3 +1125,417 @@ std::pair<std::string, std::string> FrameStore::GetManagedTypeName(ICorProfilerI
     // need to split to get the namespace and type name
     return std::make_pair(typeName.substr(0, pos), typeName.substr(pos + 1));
 }
+
+
+void FixGenericSyntax(WCHAR* name)
+{
+    ULONG currentCharPos = 0;
+    while (name[currentCharPos] != WStr('\0'))
+    {
+        if (name[currentCharPos] == WStr('`'))
+        {
+            // skip `xx
+            name[currentCharPos] = WStr('\0');
+            return;
+        }
+        currentCharPos++;
+    }
+}
+
+void FixGenericSyntax(char* name)
+{
+    ULONG currentCharPos = 0;
+    while (name[currentCharPos] != '\0')
+    {
+        if (name[currentCharPos] == '`')
+        {
+            // skip `xx
+            name[currentCharPos] = '\0';
+            return;
+        }
+        currentCharPos++;
+    }
+}
+
+void StrAppend(char* buffer, const char* str, size_t& cchBuffer)
+{
+    size_t bufLen = strlen(buffer) + 1;
+    if (bufLen <= cchBuffer)
+    {
+#ifdef _WINDOWS
+        strncat_s(buffer, cchBuffer, str, cchBuffer - bufLen);
+#else
+        strcat(buffer, str);
```

Comment:
you can use `strncat` on linux, at least we will control the number of bytes to write.

Reply:
I'm completely rewriting the string concatenation code to avoid any crash (i.e., using stringstream)